### ISSUE TITLE
Launch template with UEFI boot type doesn't work at first time

### ISSUE THREAD

**Issue by Firxiao** [apache/cloudstack#6516]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME
~~~
vm provision 
~~~

##### CLOUDSTACK VERSION

~~~
Cloudstack 4.16.1.0
vSphere Client version 7.0.2.00000
~~~

##### CONFIGURATION
~~~
Cloudstack + vSphere  
~~~

##### STEPS TO REPRODUCE

~~~
1. Create a template with UEFI boot type.
2. Create an instance via Cloudstack UI with the template(without specified boot type).
3. Check the Firmware setting in Boot Options on vSphere side, it shows "BIOS"
4. Shutdown the instance via Cloudstack UI then start it again.
5. Check the Firmware setting in Boot Options on vSphere side, it shows "UEFI"
~~~

##### EXPECTED RESULTS

~~~
Template UEFI setting should overwrite the default boot type at first time.
~~~

##### ACTUAL RESULTS

~~~
Launch UEFI Template doesn't work at first time, need to stop the instance and start it again, then it will be reconfigured with UEFI boot type.
~~~

**Comment by rohityadavcloud** [comment]
@Firxiao can you try 4.17.0.0? cc @shwstppr

**Comment by Pearl1594** [comment]
@Firxiao this has been fixed in 4.17.0 - https://github.com/apache/cloudstack/pull/5990

**Comment by rohityadavcloud** [comment]
Thanks for checking @Pearl1594, @Firxiao I'm closing this as fixed in #5990. Please try 4.17.0.0 and re-open if you're still able to reproduce this issue in 4.17.0.0. Thanks for your bug-report.
